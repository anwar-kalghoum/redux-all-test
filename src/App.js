import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="a">
        <p className="as">aaaa</p>
        <div className="in-bar">anwar</div>
      </div>
    );
  }
}

export default App;
