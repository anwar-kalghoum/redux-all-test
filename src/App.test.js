import React from 'react';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import App from './App';
 
describe('<App />', () => {
  it('renders three <App /> components', () => {
    const wrapper = shallow(<App />);
    // console.log(wrapper.debug());// console log() DOM
    expect(wrapper).toMatchSnapshot();// Snapshot Test
    console.log('ddee', expect(wrapper).toBeTruthy());
    expect(wrapper).toBeTruthy();
    expect(wrapper.find('.as').text()).toEqual('aaaa');
    expect(wrapper.find('.as').text()).toEqual('aaaa');
    expect(wrapper.find('.as').text()).toBe('aaaa');
    expect(wrapper.find('.as')).toBeTruthy();
    expect(wrapper.find('.as').text()).toHaveLength(4);
    expect(wrapper.find('.as').not('.a').text()).toHaveLength(4);
    expect(wrapper.find('p').children().text()).toHaveLength(4);
    expect(wrapper.find('.in-bar').text()).toHaveLength(5);
    expect(wrapper.find('.in-bar').text()).toHaveLength(5);
  });
});