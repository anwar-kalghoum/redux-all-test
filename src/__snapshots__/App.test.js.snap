// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders three <App /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="as"
        >
          aaaa
        </p>,
        <div
          className="in-bar"
        >
          anwar
        </div>,
      ],
      "className": "a",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "aaaa",
          "className": "as",
        },
        "ref": null,
        "rendered": "aaaa",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "anwar",
          "className": "in-bar",
        },
        "ref": null,
        "rendered": "anwar",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="as"
          >
            aaaa
          </p>,
          <div
            className="in-bar"
          >
            anwar
          </div>,
        ],
        "className": "a",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aaaa",
            "className": "as",
          },
          "ref": null,
          "rendered": "aaaa",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "anwar",
            "className": "in-bar",
          },
          "ref": null,
          "rendered": "anwar",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
